lib/kcerts/signer_test.go

```diff
@@ -0,0 +1,42 @@
+package kcerts_test
+
+import (
+	"fmt"
+	"github.com/enfabrica/enkit/lib/kcerts"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/crypto/ssh"
+	"reflect"
+	"testing"
+	"time"
+)
+
+var tableTestTypes = []kcerts.SSHKeyGenerator{kcerts.GenerateED25519, kcerts.GenerateRSA}
+
+// TestSha256Signer_PublicKey tests all possible combinations of supported kcerts.PrivateKey signing ssh.PublicKeys
+// It will sign the following ssh certs with the custom algos by their providers
+func TestSha256Signer_PublicKey(t *testing.T) {
+	for _, sourceType := range tableTestTypes {
+		for _, toSignType := range tableTestTypes {
+			_, sourcePrivKey, err := sourceType()
```

Comment:
the idiomatic way for looping over a test to try a bunch of combinations is to use `subtests`, eg, in the innermost loop, have:
```
t.Run(fmt.Sprintf("Source:%v,Sign:%v", sourceType, toSignType"), func (t *testing.T) {
   // code of your test
})
```
there's a number of reasons why this is better, see https://blog.golang.org/subtests